### ISSUE TITLE
Running the role over SSH

### ISSUE THREAD

**Issue by fubarhouse** [fubarhouse/ansible-role-macdock#10]
On behalf of Matthéo, I am opening an issue for remote execution to macOS hosts of this role. It can be achieved to a point where dockutil isn't available.

One possible solution would be to abstract this into an Ansible module.

I would hope that this could be effectively tested, ideally as well as my other roles but Apple being Apple that simply isn't possible...

I'm open to ideas, suggestions and PR's to progress this interesting feature!

**Comment by mgeoffray** [comment]
Thanks for rising my issue!  

More informations on the remote machine I can execute Dockutil - installed by Homebrew during the Ansible script - in a ZSH shell but if I run `/bin/sh dockutil` as Ansible does the command is not found.     
I don't really want to modify the `executable = /bin/sh` option in `ansible.cfg` file.

**Comment by fubarhouse** [comment]
Out of curiosity, would the ability to provide an absolute path to the executable for dockutil and a flag to enable/disable dockutil be enough for you to proceed with the role as normal?

That isn't exactly what I thought, however it may solve your problem.

If it sounds like it'll work, let me know and I can start working on a new branch for this.

**Comment by fubarhouse** [comment]
#11 Provides an installation path (rather than being provided by homebrew).

You can specify an installation path, and should fix this one...

I'll be merging that one later, but if it doesn't help I'm interested in knowing.

**Comment by mattheo-geoffray** [comment]
Hi   
Good idea! I will try this one, I'll keep you in touch today or tomorrow with this solution    
Thanks

**Comment by fubarhouse** [comment]
@mattheo-geoffray,

I've just adjusted all the paths to use absolute paths to the value of the new variable. The role should no longer depend on shell environment variables such as `$PATH` or homebrew.

When you do check, just make sure you grab the latest one! 👍

**Comment by mattheo-geoffray** [comment]
Yes I see that I am actually checking with all the changes I make a return soon!

**Comment by mattheo-geoffray** [comment]
Here is my test results with the new solution providing the full path from source (not from homebrew):  

#### When there is no other installation of dockutil    
- the initial error is solved it can go through all process 🎉 

#### When there is another installation of dockutil (e.g. Homebrew)      
- there is an error on the binary
```
"msg": "Destination /usr/local/bin/dockutil is not writable"
```    
I think we just have to check if the binary not already exist and remove it if it is

**Comment by fubarhouse** [comment]
@mattheo-geoffray,

Here's the beauty part of this change, you can now specify the path which dockutil will be installed to. You simply need to set `dockutil_install_path` to the path where you want to install it... such as `/sbin/dockutil`, `/usr/bin/dockutil` or somewhere that exists and is writable.

Be sure to include the executable file at the end (ie `dockutil`).

**Comment by fubarhouse** [comment]
@mattheo-geoffray,

You're welcome to test branch `issue-11-v3` and report back.

**Comment by mattheo-geoffray** [comment]
@fubarhouse 
yes you're right.  
Just tested and it works perfectly to me!

**Comment by fubarhouse** [comment]
@mattheo-geoffray,

That's merged, v2.2.0 has already made it to the galaxy.

Closing, but please reopen this if you feel there's more work.

Also, thanks for reaching out!
